### PR TITLE
Migrate newman tests to python

### LINKED ISSUE #663: Migrate newman integration tests to python integration tests
Too much overlap between function and intent between these two suites. Different language, and difficulty with reviewing postman export diffs make the newman suite the odd man out.

Status at : https://github.com/scitran/core/blob/migrate-newman-tests-to-python/test/integration_tests/README.md

### PR DESCRIPTION
**Pull request by ambrussimon-invenshure**
### Review Checklist

- Tests were added to cover all code changes
- Documentation was added / updated
- Code and tests follow standards in CONTRIBUTING.md

Resolves  #663

### CODE REVIEW

**Comment by ambrussimon-invenshure** [comment]
I covered every request @ryansanford and sqashed the PR, too. I hope squashing it doesn't hinder reviewability, @nagem.

**Comment by nagem** [comment]
>60.745%

🎉🎉🎉🎉

**Comment by nagem** [comment]
Nice work, @ambrussimon-invenshure! I'll review this as I get time today and tomorrow.

File: TESTING.md [hunk 1]
```diff
@@ -19,28 +19,6 @@
 
 ### Tools
 - [abao](https://github.com/cybertk/abao/)
-- [postman](https://www.getpostman.com/docs/)
 
 ### Testing API against RAML with Abao
 Abao is one of the testing tools run during our TravisCI build.  It tests the API implementation against what’s defined in the RAML spec.  Adding a new resource / url to the RAML spec will cause Abao to verify that resource during integration tests.  Sometimes abao cannot properly test a resource (file field uploads) or a test may require chaining variable.  Abao has before and after hooks for tests, written in javascript.  These can be used to skip a test, inject variables into the request, or make extra assertions about the response.  See tests/integration/abao in the repo for the hooks file.  See [abao github readme](https://github.com/cybertk/abao/blob/master/README.md) for more information on how to use hooks.
```
Comment (ryansanford) [inline TESTING.md]: I like the new paragraph, but leave some room for misinterpretation. I'd suggest this alteration:

Abao tests can depend on specific resources (eg. group, project, session, etc.) pre-existing in the DB. That resource loading should be maintained within`test/integration_tests/abao/load_fixture.py` and is executed automatically via the integration test scripts within `test/bin`.

**Comment by ryansanford** [comment]
@ambrussimon-invenshure Added new comment related to new abao paragraph. Everything else looks good on my end.

File: test/integration_tests/abao/load_fixture.py [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+    })
+    assert r.ok
+    assert r.json()['modified'] == 1
+
+    # test session 1 is marked as compilant
+    # depends on 'create compliant session'
+    r = session.get('/sessions/' + st_compliant_session['_id'])
+    assert r.ok
+    assert r.json()['project_has_template'] == True
+    assert r.json()['satisfies_template'] == True
```
Comment (kofalt) [inline test/integration_tests/abao/load_fixture.py]: From this point onwards, you have a few tests that don't have to do with just inserting the data. Does it make sense to keep them here for convenience? Do you think they might move to other files in the future?

File: test/integration_tests/python/test_containers.py [hunk 3]
```diff
@@ -85,3 +85,44 @@ def test_switching_acquisition_between_projects(with_two_groups, data_builder, a
...
+        'acquisitions': [{ 'label': '_testing$', 'minimum': 2 }]
+    })
+    assert r.ok
+    assert r.json()['modified'] == 1
+
+    # test non-compliant session (wrong subject.code and #acquisitions)
+    r = as_user.get('/sessions/' + data.session)
+    assert r.ok
+    assert r.json()['project_has_template'] is True
+    assert r.json()['satisfies_template'] is False
```
Comment (kofalt) [inline test/integration_tests/python/test_containers.py]: @nagem, just a quick sanity check, do you want `is True` or `== True`?

Deferring to you, maybe pep-213092841 has opinions, etc.

**Comment by kofalt** [comment]
This looks great. Excellent work!

I have no major objections. Bouncing to @nagem.

File: test/integration_tests/python/test_containers.py (hunk 3 above)
Comment (ambrussimon-invenshure) [inline test/integration_tests/python/test_containers.py]: Ah, found the [PEP recommendation](https://www.python.org/dev/peps/pep-0008/#programming-recommendations) (At the very end of the section, CTRL+F "if greeting"):

> 
> Don't compare boolean values to True or False using == .
> 
> Yes:   if greeting:
> No:    if greeting == True:
> Worse: if greeting is True:
> 

For now I will just change from 'is' to '==' to make the usage consistent within the project.
Nice catch @kofalt, thanks!

File: test/integration_tests/abao/load_fixture.py (hunk 2 above)
Comment (ambrussimon-invenshure) [inline test/integration_tests/abao/load_fixture.py]: The coverage was already "duplicated" in python tests, so these below should definitely go away as they aren't needed. Thanks!

File: test/integration_tests/python/states.py [hunk 4]
```diff
@@ -30,5 +24,21 @@ def teardown_db():
     fixture_data.project = project
     fixture_data.session = session
     fixture_data.acquisition = acquisition
+    return fixture_data
+
+
+# NOTE removing scope='module' form with_hierarchy_and_file_data fixed the need
+# for keeping the duplicate in test_uploads. Now the fixture runs per test,
+# eliminating any data dependency between tests within a module at the price
+# of more load/teardown requests.
```
Comment (nagem) [inline test/integration_tests/python/states.py]: 👍

**Comment by nagem** [comment]
Great work! Everything looks good to me.

**Pull request merged**